archive/logic_puzzles/lady_or_tiger.lean

```diff
@@ -0,0 +1,235 @@
+/-
+Copyright (c) 2020 Dan Stanescu.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Dan Stanescu and Yuri G. Kudryashov.
+-/
+
+import tactic
+
+/-!
+# Six logic puzzles.
+-/
+
+/-- 
+The first six puzzles from:
+    "The Lady or the Tiger? And Other Logic Puzzles"
+        by Raymond Smullyan.
+First three contributed to the Lean Zulip chat by Yury G. Kudryashov
+but apparently set up by his seven-years-old son.
+Slightly modified in appearance (for readability) but not in content by D. Stanescu.
+
+A king has prisoners choose between two rooms. The signs on the room doors offer enough information
+to allow them to make the right choice between rooms with ladies (i.e. presumably happiness) 
+and rooms with tigers (i.e. presumably death).
+In the first three puzzles, the king explains that it is always the case that one of the door 
+signs is true while the other one is false.
+-/
+
+
+inductive door_leads_to
+| lady
+| tiger
+
+notation `y` := door_leads_to.lady
+notation `n` := door_leads_to.tiger
+
+structure Q := (d₁ d₂ : door_leads_to)
+
+/--
+First puzzle: 
+Sign on the first door indicates that there's a lady in this room and a tiger in the other room.
+Sign on second door indicates that in one of the rooms there's a lady and in one room a tiger.
+-/
+
+def Q1 := Q
+
+namespace Q1
+
+variables (q : Q1)
+
+-- Sign on first door
+def D1 := q.1 = y ∧ q.2 = n
+-- Sign on second door
+def D2 := (q.1 = y ∨ q.2 = y) ∧ (q.1 = n ∨ q.2 = n)
+
+def H := q.D1 ∧ ¬q.D2 ∨ ¬q.D1 ∧ q.D2
+
+lemma answer_term : q.H → q.1 = n ∧ q.2 = y :=
+  by rcases q with ⟨_|_,_|_⟩; simp [H, D1, D2]
```

Comment:
This is still a tactic mode proof! All you've done is used `by` instead of `begin ... end`.

Reply:
True.

Reply:
Will change my comment.

Reply:
The declaration name also?

Reply:
Totally, my omission!